tools/download.py

```diff
@@ -0,0 +1,90 @@
+import argparse
+import os
+from itertools import repeat
+from multiprocessing.pool import ThreadPool
+from pathlib import Path
+from zipfile import ZipFile
+
+import torch
+
+
+def download(url, dir, unzip=True, delete=False, curl=False, threads=1):
+
+    def download_one(url, dir):
+        f = dir / Path(url).name
+        if Path(url).is_file():
+            Path(url).rename(f)
+        elif not f.exists():
+            print('Downloading {} to {}'.format(url, f))
+            if not curl:
+                os.system('curl -L {} -o {} --retry 9 -C -'.format(url, f))
+            else:
+                torch.hub.download_url_to_file(url, f, progress=True)
+        if unzip and f.suffix in ('.zip', '.gz'):
+            print('Unzipping {}'.format(f.name))
+            if f.suffix == '.zip':
+                ZipFile(f).extractall(path=dir)
+            elif f.suffix == '.gz':
+                os.system(f'tar xfz {f} --directory {f.parent}')
+            if delete:
+                f.unlink()
+
+    dir = Path(dir)
+    dir.mkdir(parents=True, exist_ok=True)
+    if threads > 1:
+        pool = ThreadPool(threads)
+        pool.imap(lambda x: download_one(*x), zip(url, repeat(dir)))
+        pool.close()
+        pool.join()
+    else:
+        for u in [url] if isinstance(url, (str, Path)) else url:
+            download_one(u, dir)
+
+
+def main(args):
+    path = Path(args.save_dir)
+    if not path.exists():
+        path.mkdir(parents=True, exist_ok=True)
+
+    urls = [
```

Comment:
Suggest putting the urls to a dict because we will also download VOC/LVIS dataset in the future, you can put them in a dict like below:

```python
data2url = dict(coco2017=[
    URLs...
], lvis=[], voc2007=[], )

```